rust/datafusion/src/physical_plan/type_coercion.rs

```diff
@@ -68,6 +68,29 @@ pub fn data_types(
     current_types: &[DataType],
     signature: &Signature,
 ) -> Result<Vec<DataType>> {
+    let valid_types = get_valid_types(signature, current_types)?;
+
+    if valid_types.contains(&current_types.to_owned()) {
```

Comment:
Why can't this be `&current_types` (aka why does it need a call to `to_owned` just to immediately borrow from it?)

Reply:
Will have a look... I think it was auto generated by the new "extract function" functionality in rust-analyzer (which doesn't work 100% reliably, but still is very useful).

Reply:
seems `&current_types` isn't possible with `contains`, made it use `any`  instead.